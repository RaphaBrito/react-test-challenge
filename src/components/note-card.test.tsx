import { expect, it, vi } from "vitest";

import {
    NoteCard,
    type NoteCardProps,
  } from "./note-card";
  import { render, screen, userEvent } from "../test/utils";
  import { type Note } from "../types/note";

describe("NoteCard component", () => {
    it("should render correctly with an existing note", () => {
        const note = {
            id: 0,
            title: "note title",
            description: "note description"
          } satisfies Note;

        const props = {
        note,
          onEdit: () => undefined,
          onDelete: () => undefined,
        } satisfies NoteCardProps;
      
        const { container } = render(<NoteCard {...props} />); 

        expect(container).toMatchSnapshot();
      });

    it("when 'Editar' button is clicked, it should disappear", async() => {
        const onEdit = vi.fn()

        const note = {
            id: 0,
            title: "note title",
            description: "note description"
        } satisfies Note;

        const props = {
            note,
            onEdit,
            onDelete: () => undefined,
        } satisfies NoteCardProps;

        await userEvent.click(screen.getByText("Editar"));

        expect(screen.queryByText("Editar")).not.toBeInTheDocument();
        expect(screen.queryByText("Salvar")).toBeInTheDocument();
        expect(screen.queryByText("Cancelar")).toBeInTheDocument();
    });

  it("should call onDelete when 'Remover' button is clicked", async() => {
    const onDelete = vi.fn()

    const note = {
        id: 0,
        title: "note title",
        description: "note description"
    } satisfies Note;

    const props = {
        note,
        onEdit: () => undefined,
        onDelete
    } satisfies NoteCardProps;

    const result = render(<NoteCard {...props} />);

    await userEvent.click(screen.getByText("Remover"));

    expect(onDelete).toHaveBeenCalledWith(note.id);
  });
});
