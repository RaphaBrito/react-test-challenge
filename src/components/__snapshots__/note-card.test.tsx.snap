// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NoteCard component > should render correctly with an existing note 1`] = `
<div>
  <div
    class="_card_277467"
  >
    <h3>
      note title
    </h3>
    <p>
      note description
    </p>
    <div
      class="_cardFooter_277467"
    >
      <button>
        Editar
      </button>
      <button>
        Remover
      </button>
    </div>
  </div>
</div>
`;
